api/README.md

```diff
@@ -91,26 +92,82 @@ watch KUBECONFIG=/tmp/kubeconfig kubectl get pod -A
+knative-serving   net-istio-controller-544874485d-mncx9     1/1     Running   0          17h
+default           mockserver-7844d797f7-zwk2b               1/1     Running   0          17h
+kube-system       local-path-provisioner-84bb864455-58c7z   1/1     Running   0          17h
+kube-system       coredns-96cc4f57d-t8vsw                   1/1     Running   0          17h
+knative-serving   domainmapping-webhook-74fc9b87b4-pmqdg    1/1     Running   0          17h
+istio-system      istiod-7c595445b6-r6fwt                   1/1     Running   0          17h
+knative-serving   activator-7d658db58b-29pnb                1/1     Running   0          17h
+kube-system       metrics-server-ff9dbcb6c-bjwl2            1/1     Running   0          17h
+istio-system      svclb-istio-ingressgateway-kxtmp          3/3     Running   0          17h
+istio-system      svclb-istio-ingressgateway-xszc7          3/3     Running   0          17h
+istio-system      svclb-istio-ingressgateway-xg6zg          3/3     Running   0          17h
+default           spark-spark-operator-59c685545-qqn44      1/1     Running   0          17h
+istio-system      svclb-istio-ingressgateway-mn9n4          3/3     Running   0          17h
+istio-system      istio-ingressgateway-b7ffbd9c6-kfpl2      1/1     Running   0          17h
+```
+
+#### <b>Prepare proprietary experiment engine</b>
+---
+Next, let's get the proprietary experiment engine plugin ready!
+
+1. Build the proprietary experiment engine plugin binary
+
+When building the binary, make sure you're using the same GOOS and GOARCH that's compatible with your machine.
+
+```bash
+cd engines/experiment/examples/plugins/hardcoded
+# If using MacOS M1
+make proprietary_exp_plugin GOOS=darwin GOARCH=arm64
+```
+
+2. Build the proprietary experiment engine and push it to the local registry.
+
+```bash
+pushd ../experiment/examples/plugins/hardcoded
+{
+    docker build -t localhost:5000/proprietary-experiment-engine-plugin .
+    docker push localhost:5000/proprietary-experiment-engine-plugin
+}
+popd
+```
+
+If you're facing issues building the proprietary experiment engine docker image locally, do the following steps.
+
+1. Replace "turing/engines/experiment/examples/plugins/hardcoded" Dockerfile with the following Dockerfile.
+```docker
+FROM alpine:3.15
+
+ARG TURING_USER="turing"
+ARG TURING_USER_GROUP="app"
+
+RUN addgroup -S ${TURING_USER_GROUP} \
+    && adduser -S ${TURING_USER} -G ${TURING_USER_GROUP} -H
+
+ENV PLUGIN_NAME ""
+ENV PLUGINS_DIR "/app/plugins"
+
+COPY --chown=${TURING_USER}:${TURING_USER_GROUP} ./bin/example-plugin /go/bin/plugin
+
+CMD ["sh", "-c", "cp /go/bin/plugin ${PLUGINS_DIR}/${PLUGIN_NAME:?variable must be set}"]
 ```
 
+2. Build and push the image to the local registry.
+
+```bash
+docker build --no-cache -t localhost:5000/proprietary-experiment-engine-plugin
+docker push localhost:5000/proprietary-experiment-engine-plugin
+```
+
+#### <b>Prepare Turing router</b>
```

Comment:
I think the image building commands below, for the router, can also be rewritten using the existing `make build-image` target for the router, to simply the manual instructions in the README.